Change timestamp format

ritzalam opened the pull request:
Currently, the timestamp format is system dependent. So when parsing for the the timestamp will sometimes fail is the format is not the expected format.

Make the timestamp format to "2021-09-23T16:08:29.000+00:00".

ritzalam committed:
- set timestamp format to "2021-09-23T16:44:46.000+00:00"

ritzalam committed:
- extract timestamp conversion into method

ritzalam committed:
- change how the format is set and add millis

kepstin reviewed (changes requested):
I don't think this approach will work as-is. The way we are using the bbb-events gem in other places expects values returned by the gem to be Time objects, so that date formatting methods can be run on them. Changing that so it returns pre-formatted strings instead breaks stuff.

kepstin commented:
I think we need to specifically override the `.to_json` methods in various places to do explicit formatting when converting dates to json.

kepstin committed:
Split json generation from rails hash generation

When generating a hash with to_h, we want to leave native ruby types for
other ruby code to consume.

For json generation, add a new `as_json` method that returns a hash
designed to be passed to the `JSON.generate` method, which performs type
conversion - timestamp formatting in particular.

ritzalam commented:
> I think we need to specifically override the `.to_json` methods in various places to do explicit formatting when converting dates to json.

This has been added.

kepstin commented on example.rb:
```diff
@@ -36,6 +37,8 @@
 
 puts "Writing the JSON data"
 
+puts JSON.pretty_generate(recording.as_json)
```
Looks like this debug output got left in.

ritzalam replied:
I left it so that we can see right away the result. The data.json file contains unformatted JSON that we have to run through a formatter.

kepstin replied:
I don't want this turned on in the production deployment, and it also causes a bunch of duplicate work since it's generating the json twice.

ritzalam replied:
This file example.rb isn't used in prod. Only in development and in the command line.

kepstin replied:
Ah, I missed that. Should be fine then.

kepstin committed:
Update Gemfile.lock
